src/page-modules/departures/utils.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  TransportModeFilterState,
+  filterToQueryString,
+} from '@atb/modules/transport-mode';
+import { FromDepartureQuery } from './types';
+import { searchTimeToQueryString } from '@atb/modules/search-time';
+import { Url } from 'url';
+import { ParsedUrlQueryInput } from 'querystring';
+
+export function createFromQuery(
+  tripQuery: FromDepartureQuery,
+  transportModeFilter: TransportModeFilterState,
+): {
+  pathname: string;
+  query: ParsedUrlQueryInput;
+} {
+  let transportModeFilterQuery = {};
+  if (transportModeFilter) {
+    const filterQueryString = filterToQueryString(transportModeFilter);
+    if (filterQueryString) {
+      transportModeFilterQuery = {
+        filter: filterQueryString,
+      };
+    }
+  }
+
+  const searchTimeQuery = searchTimeToQueryString(tripQuery.searchTime);
+
+  if (tripQuery.from?.layer == 'venue') {
+    return {
+      pathname: '/departures/[[...id]]',
+      query: {
```

Comment:
Cf the TODO comment that was in the original code regarding URL encoding; is it preferable for us to encode the URL (like it is done automatically with this method) or don't and keep some legibility? I guess the URL is so complex that legibility for the user might not be a concern.

Reply:
I don't think I get the question